frontend/src/components/Project/CreateProject.tsx

```diff
@@ -0,0 +1,561 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Button,
+  FormHelperText,
+  Grid,
+  MenuItem,
+  Paper,
+  Select,
+  styled,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import { FormikProps, useFormik } from 'formik';
+import { AxiosResponse } from 'axios';
+import { getAllClientAccount, getAllRegexEmployeeAccount } from '../../services/AccountAPI';
+import { createProject, getProject, updateProject } from '../../services/ProjectAPI';
+import createProjectFromSchema from './CreateProjectFormValidationSchema';
+import useStyles from './CreateProjectStyle';
+import Autocomplete, { AutocompleteInputChangeReason } from '@material-ui/lab/Autocomplete';
+import { useHistory } from 'react-router';
+import { SaleCreationDTO } from '../../dto/SaleDTO';
+
+interface Props {
+  id?: string;
+  edit: string;
+}
+
+const CreateProject: React.FC<Props> = ({ id, edit }) => {
+  const [created, setCreated] = useState<boolean>(false);
+  const date = new Date();
+  const history = useHistory();
+  const classes = useStyles();
+
+  interface CreateProjectFormData {
+    title: string;
+    description: string;
+    status: string;
+    serviceType: string;
+    leadSource: string;
+    leadCredit: string;
+    leadRanking: string;
+    followUpDate: string;
+    deadLineDate: string;
+    extraNotes: string;
+    email: string;
+    saleDescription: string;
+    saleValue: number | string;
+    assignee: Object[] | undefined;
+  }
+
+  interface EditFetchData {
+    id: number;
+    businessId: string;
+    createdDate: string;
+    deadlineDate: string;
+    description: string;
+    email: string;
+    extraNotes: string;
+    followUpDate: string;
+    leadCredit: string;
+    leadRanking: string;
+    leadSource: string;
+    sale: SaleCreationDTO;
+    serviceType: string;
+    status: string;
+    title: string;
+  }
+
+  const formik: FormikProps<CreateProjectFormData> = useFormik<CreateProjectFormData>({
+    onReset: () => {},
+    initialValues: {
+      title: '',
+      description: '',
+      status: '',
+      serviceType: '',
+      leadSource: '',
+      leadCredit: '',
+      leadRanking: '',
+      followUpDate: date.toISOString().split('T')[0],
+      deadLineDate: date.toISOString().split('T')[0],
+      extraNotes: '',
+      email: '',
+      saleDescription: '',
+      saleValue: '',
+      assignee: [],
+    },
+    onSubmit: async (values) => {
+      if (editState === 'true') {
+        submitEdit();
+      } else {
+        const response: AxiosResponse<any> = await createProject({
```

Comment:
should wrap this in try catch for fail case

Reply:
ok